File: 프로그래머스/2/49993. 스킬트리/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    skilllist = list(skill)
    
    for i in range(len(skill_trees)) :
        skillstack = skilllist.copy()
        checkskill = True
        
        for j in range(len(skill_trees[i])) :
            
            if skill_trees[i][j] in skill :
                #있긴 있다
                if skillstack[0] == skill_trees[i][j] :
                    #잘맞는다
                    skillstack.pop(0)
                    if len(skillstack) == 0 :
                       break
                
                else: #있긴한데, 번호가 안맞다.
                    checkskill = False
                    break
        
        if checkskill == True:
                answer+=1

                
        
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
import java.util.*;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        
        Queue<Character> q = new LinkedList<>();
        for (int i = 0 ; i<skill.length(); i++)
        {
            q.add(skill.charAt(i));
        }
        
        for (int i = 0 ; i<skill_trees.length; i++)
        {
            String nowstr = skill_trees[i];
            int nowcompare = 0;
            
            boolean flag = true;
            for (int j = 0 ; j<nowstr.length(); j++)
            {
                if (!q.contains(nowstr.charAt(j)))
                {
                    continue;
                }
                
                if (q.peek() == nowstr.charAt(j))
                {
                    q.poll();
                }
                else
                {
                    flag = false;
                    break;
                }
            }
            
            if (flag)
            {
                answer +=1;
            }
            
             q = new LinkedList<>();
             for (int j = 0 ; j<skill.length(); j++)
             {
                 q.add(skill.charAt(j));
             }
        
        }
        
        return answer;
    }
}